shell/common/switches.cc

```diff
@@ -41,8 +41,8 @@ struct SwitchDesc {
 
 // clang-format off
 static const std::string gDartFlagsWhitelist[] = {
-    "--no-causal_async_stacks",
     "--lazy_async_stacks",
+    "--no-causal_async_stacks",
```

Comment:
Let's add a test to `runtime_unittests` (specifically `dart_vm_unitttests.cc`) that verifies whitelists. Those work in AOT as well as JIT.

Reply:
> Since this is now already this default, putting it in the whitelist doesn't make sense anymore.
> The Dart whitelist can be empty in release mode now

Some flutter downstream customers might explicitly specify them, which is why I left it here.

> To prevent breaking workflows that specify the flag that was in the whitelist but isn't anymore because it is the default, 
> you'll need to patch IsWhitelistedDartVMFlag to not terminate VM launch if a non-whitelisted but default flag is 
> specified by the embedder.

Doesn't this have the same effect as just leaving it in the whitelist?

This is only temporary anyway, the plan is as follows:

**Step 1)** Enable it in JIT and AOT by-default in the engine & roll it through (this PR)
**Step 2)** Change the Dart VM defaults & roll it through (cannot do this right now, because embedders need to explicitly specify both flags for that to work)
**Step 3)** Remove both flags in all places (g3, flutter, dart vm)

I plan on doing these three steps in the next 3 weeks to give each change some time to roll.

Reply:
> Doesn't this have the same effect as just leaving it in the whitelist?

Right. It was just a suggestion for cleaning up the code a bit for readability. I was wondering why the flag was specified in a whitelist if it was already applied by default. Folks might try to "clean it up" in the future and break exiting apps. Hence my suggestion to just make the whitelist check just ignore the default flags and reject additional flags not in the whitelist.

If this is temporary, go for it.

> Step 3) Remove both flags in all places (g3, flutter, dart vm)

This still leaves custom embedders susceptible to having VM launches terminated by the flags not in the whitelist (but present by default). For this reason, I believe updating the whitelist check is a less invasive way to go.

Reply:
> This still leaves custom embedders susceptible to having VM launches terminated by the flags not in the whitelist (but 
> present by default). For this reason, I believe updating the whitelist check is a less invasive way to go.

Thank you for explaining, that makes sense. I can do that change when I remove the explicit passing of the flags in Step 3 (which will first be done in the engine, only later on we'll remove the flag support from the VM itself). Does that sound ok?